File: result/metagpt/claude/humaneval/120.c
#include <stdio.h>
#include <stdlib.h>

// Function to compare two integers (used for qsort)
int compare(const void *a, const void *b) {
    return (*(int*)a - *(int*)b);
}

// Function to find the maximum k numbers in the array and return them in a sorted array
int* maximum(int* arr, int arrSize, int k, int* returnSize) {
    // Sort the array in ascending order
    qsort(arr, arrSize, sizeof(int), compare);
    
    // Allocate memory for the output array
    int* out = (int*)malloc(k * sizeof(int));
    
    // Copy the last k elements from the sorted array to the output array
    for (int i = 0; i < k; i++) {
        out[i] = arr[arrSize - k + i];
    }
    
    // Set the return size
    *returnSize = k;
    
    return out;
}

File: result/metagpt/gpt4o/humaneval/120.rs
fn compare(a: &i32, b: &i32) -> std::cmp::Ordering {
    a.cmp(b)
}

fn maximum(arr: &mut [i32], k: usize) -> Vec<i32> {
    // Sort the array in ascending order
    arr.sort_by(compare);

    // Collect the last k elements from the sorted array
    arr[arr.len() - k..].to_vec()
}

fn main() {
    let mut arr = vec![3, 1, 4, 1, 5, 9, 2, 6, 5, 3, 5];
    let k = 3;
    let result = maximum(&mut arr, k);
    println!("Maximum {} numbers: {:?}", k, result);
}